### PR TITLE
Add broken link check workflow

### PR DESCRIPTION
**Pull request by ohltyler**
Signed-off-by: Tyler Ohlsen <ohltyler@amazon.com>

### Description
Adds a broken link checker workflow. Runs on pushes & pull requests to `main`. Based off of the `opensearch-plugins` [PR](https://github.com/opensearch-project/opensearch-plugins/pull/54).
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by kaituo — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [![AD Test](https://github.com/opensearch-project/anomaly-detection/workflows/Build%20and%20Test%20Anomaly%20detection/badge.svg)](https://github.com/opensearch-project/anomaly-detection/actions?query=workflow%3A%22Build+and+Test+Anomaly+detection%22+branch%3A%22main%22)
 [![codecov](https://codecov.io/gh/opensearch-project/anomaly-detection/branch/main/graph/badge.svg?flag=plugin)](https://codecov.io/gh/opensearch-project/anomaly-detection)
-[![Documentation](https://img.shields.io/badge/doc-reference-blue)](https://opendistro.github.io/for-elasticsearch-docs/docs/ad/)
+[![Documentation](https://img.shields.io/badge/doc-reference-blue)](https://docs-beta.opensearch.org/monitoring-plugins/ad/index/)
```
Comment (ylwu-amzn) [inline README.md:3]: Does `lychee` support a deny list of regex, for example any link starting with  `https://opendistro.github.io` will be invalid.

Comment (ohltyler) [reply README.md:3]: From [here](https://github.com/lycheeverse/lychee#commandline-parameters), don't believe so, although that could be helpful.

Comment (ylwu-amzn) [reply README.md:3]: Got it. I created an issue to ask deny list feature: https://github.com/lycheeverse/lychee/issues/267

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Pull request merged**